src/main/java/org/zalando/nakadi/service/subscription/zk/AbstractZkSubscriptionClient.java

```diff
@@ -122,48 +115,28 @@ public final void deleteSubscription() {
         }
     }
 
-    /**
-     * Checks that version of topology/offsets representation is the same as expected.
-     *
-     * @param bytes
-     * @return true, if its same, false - if not, but can be migrated.
-     * @throws ForwardVersionNotAllowed In case of version can not be migrated.
-     */
-    protected abstract boolean checkTopologyVersion(byte[] bytes) throws ForwardVersionNotAllowed;
-
     @Override
-    public final boolean isSubscriptionCreatedAndInitialized() throws ForwardVersionNotAllowed {
+    public final boolean isSubscriptionCreatedAndInitialized() throws NakadiRuntimeException {
         // First step - check that state node was already written
         try {
             final String state = new String(getCurator().getData().forPath(getSubscriptionPath("/state")), UTF_8);
             if (!state.equals(STATE_INITIALIZED)) {
                 return false;
             }
+            return true;
         } catch (final KeeperException.NoNodeException ex) {
             return false;
         } catch (final Exception e) {
             throw new NakadiRuntimeException(e);
         }
-        // Second step - check that version is supported.
-        try {
-            return checkTopologyVersion(getCurator().getData().forPath(getSubscriptionPath(NODE_TOPOLOGY)));
-        } catch (final RuntimeException e) {
-            throw e;
-        } catch (final Exception e) {
-            throw new NakadiRuntimeException(e);
-        }
     }
 
     @Override
-    public final void fillEmptySubscription(final Collection<SubscriptionCursorWithoutToken> cursorsIn) {
+    public final void fillEmptySubscription(final Collection<SubscriptionCursorWithoutToken> cursors) {
         try {
-            final Collection<SubscriptionCursorWithoutToken> cursors;
-            // Delete root subscription node
+            // Delete root subscription node, if it was erroneously created
```

Comment:
When initialization didn't finish (for example initialization started, but then someone killed nakadi with -9)